src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/ExecutionManager/ExecutionManagerBase.ReadyToRunJitManager.cs

```diff
@@ -0,0 +1,136 @@
+                return false;
+
+            // Find the relative address that we are looking for
+            TargetCodePointer code = CodePointerUtils.CodePointerFromAddress(jittedCodeAddress.AsTargetPointer, Target);
+            TargetPointer imageBase = rangeSection.Data.RangeBegin;
+            TargetPointer relativeAddr = code - imageBase;
+
+            int index = GetRuntimeFunctionIndexForAddress(r2rInfo, relativeAddr);
+            if (index < 0)
+                return false;
+
+            bool featureEHFunclets = Target.ReadGlobal<byte>(Constants.Globals.FeatureEHFunclets) != 0;
+            if (featureEHFunclets)
+            {
+                // TODO: [cdac] Look up in hot/cold mapping lookup table and if the method is in the cold block,
+                // get the index of the associated hot block.
+                //   HotColdMappingLookupTable::LookupMappingForMethod
+                //
+                // while GetMethodDescForEntryPoint for the begin address of function at index is null
+                //   index--
+            }
+
+            TargetPointer functionEntry = r2rInfo.RuntimeFunctions + (ulong)(index * _runtimeFunctionSize);
+            Data.RuntimeFunction function = Target.ProcessedData.GetOrAdd<Data.RuntimeFunction>(functionEntry);
+
+            // ReadyToRunInfo::GetMethodDescForEntryPointInNativeImage
+            TargetPointer startAddress = imageBase + function.BeginAddress;
+            TargetPointer entryPoint = CodePointerUtils.CodePointerFromAddress(startAddress, Target).AsTargetPointer;
+
+            TargetPointer methodDesc = _lookup.GetValue(r2rInfo.EntryPointToMethodDescMap, entryPoint);
+            Debug.Assert(methodDesc != TargetPointer.Null);
+
+            // TODO: [cdac] Handle method with cold code when computing relative offset
+            // ReadyToRunJitManager::JitTokenToMethodRegionInfo
+            TargetNUInt relativeOffset = new TargetNUInt(code - startAddress);
+
+            info = new CodeBlock(startAddress.Value, methodDesc, relativeOffset, rangeSection.Data!.JitManager);
+            return true;
+        }
+
+        private bool IsStubCodeBlockThunk(Data.RangeSection rangeSection, Data.ReadyToRunInfo r2rInfo, TargetCodePointer jittedCodeAddress)
+        {
+            if (r2rInfo.DelayLoadMethodCallThunks == TargetPointer.Null)
+                return false;
+
+            // Check if the address is in the region containing thunks for READYTORUN_HELPER_DelayLoad_MethodCall
+            Data.ImageDataDirectory thunksData = Target.ProcessedData.GetOrAdd<Data.ImageDataDirectory>(r2rInfo.DelayLoadMethodCallThunks);
+            ulong rva = jittedCodeAddress - rangeSection.RangeBegin;
+            return thunksData.VirtualAddress <= rva && rva < thunksData.VirtualAddress + thunksData.Size;
+        }
+
+        private int GetRuntimeFunctionIndexForAddress(Data.ReadyToRunInfo r2rInfo, TargetPointer relativeAddress)
```

Comment:
Should we assert the Thumb case here? I see in the original it is applied for this case.